elastic/kibana
# [CTI] explicit threat_indicator_path for IM rules

## Summary

1. adds threat_indicator_path explicit default value to create rule UI
2. adds threat_indicator_path explicit default value to create rule api

## Testing

1. when creating an indicator match rule using the UI
    1. Confirm that the prepopulated value of About Step / advanced settings / Indicator prefix override is `threat.indicator`
    2. Confirm that user can't reach the next step due to validation error when the Indicator prefix override value is empty
    3. Confirm that user can reach the next step with a custom value for Indicator prefix override, which correctly persists when the rule is saved
    4. Confirm that user can create an IM rule without modifying advanced settings, and the `threat_indicator_path` value on the saved rule is `threat.indicator`

2. when creating any other rule using the UI (for instance, a custom query rule)
    1. Confirm that user can create the rule as expected, and there is no value for `threat_indicator_path` on the rule once it's saved.

3. when creating an indicator match rule using the API
    1. Open the Network tab in Chrome Dev Tools and create an IM rule. Copy the create rule request as cURL and edit the to remove `threatIndicatorPath` from the body. Make the request and confirm that the created rule has a `threat_indicator_path` field with the value `threat.indicator`

4. when creating any other rule using the API
    1. Open the Network tab in Chrome Dev Tools and create a rule (that is not an indicator match rule). Copy the create rule request. Make the request and confirm that the created rule does not have `threat_indicator_path` field.

### Checklist

Delete any items that are not applicable to this PR.



### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to identify risks that should be tested prior to the change/feature release.

When forming the risk matrix, consider some of the following examples and how they may potentially impact the change:

| Risk                      | Probability | Severity | Mitigation/Notes        |
|---------------------------|-------------|----------|-------------------------|
| Multiple Spaces&mdash;unexpected behavior in non-default Kibana Space. | Low | High | Integration tests will verify that all features are still supported in non-default Kibana Space and when user switches between spaces. |
| Multiple nodes&mdash;Elasticsearch polling might have race conditions when multiple Kibana nodes are polling for the same tasks. | High | Low | Tasks are idempotent, so executing them multiple times will not result in logical error, but will degrade performance. To test for this case we add plenty of unit tests around this logic and document manual testing procedure. |
| Code should gracefully handle cases when feature X or plugin Y are disabled. | Medium | High | Unit tests will verify that any feature flag or plugin combination still results in our service operational. |
| [See more potential risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx) |


### For maintainers
